p2p/src/authenticated/discovery/actors/listener.rs

```diff
@@ -33,34 +40,56 @@ pub struct Actor<
 
     address: SocketAddr,
     stream_cfg: StreamConfig<C>,
-    rate_limiter: RateLimiter<NotKeyed, InMemoryState, E, NoOpMiddleware<E::Instant>>,
-
-    handshakes_rate_limited: Counter,
+    handshake_limiter: Limiter,
+    ip_rate_limiter: RateLimiter<IpAddr, HashMapStateStore<IpAddr>, E, NoOpMiddleware<E::Instant>>,
+    subnet_rate_limiter:
+        RateLimiter<Subnet, HashMapStateStore<Subnet>, E, NoOpMiddleware<E::Instant>>,
+    handshakes_concurrent_rate_limited: Counter,
+    handshakes_ip_rate_limited: Counter,
+    handshakes_subnet_rate_limited: Counter,
 }
 
 impl<E: Spawner + Clock + ReasonablyRealtime + Network + Rng + CryptoRng + Metrics, C: Signer>
     Actor<E, C>
 {
     pub fn new(context: E, cfg: Config<C>) -> Self {
         // Create metrics
-        let handshakes_rate_limited = Counter::default();
+        let handshakes_concurrent_rate_limited = Counter::default();
+        context.register(
+            "handshake_concurrent_rate_limited",
+            "number of handshakes dropped because maximum concurrent handshakes was reached",
```

Comment:
nit to make this parallel with other metric descriptions

```suggestion
            "number of handshake attempts dropped because maximum concurrent handshakes was reached",
```